hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -188,6 +185,45 @@ def process_current_session(self, settings, parent_item):
         # reset the original document to restore the state for the user
         engine.adobe.app.activeDocument = current_document
 
+    def _set_item_thumbnail_for_document(self, document_item, document):
+        """
+        Creates thumbnail for photoshop document.
+        Input:
+        document_item: Item associated to photoshop document
+        document: Photoshop document.
+        """
+
+        publisher = self.parent
+        engine = publisher.engine
+        path = _document_path(document)
+        if path:
```

Comment:
suggest returning here if no path to avoid indentation in the rest of the method. Also suggest updating the docstring to indicate what happens if the file has not been saved.